includes/Core/Util/URL.php

```diff
@@ -147,13 +149,13 @@ public static function permute_site_hosts( $hostname ) {
 			// An ASCII hostname can only be non-IDN or punycode-encoded.
 			if ( $is_ascii ) {
 				// If the hostname is in punycode encoding, add the decoded version to the list of hosts.
-				if ( 0 === strpos( $hostname, Punycode::PREFIX ) || false !== strpos( $hostname, '.' . Punycode::PREFIX ) ) {
-					$host_decoded = $punycode->decode( $hostname );
+				if ( 0 === strpos( $hostname, self::PUNYCODE_PREFIX ) || false !== strpos( $hostname, '.' . self::PUNYCODE_PREFIX ) ) {
+					$host_decoded = idn_to_utf8( $hostname, IDNA_DEFAULT, INTL_IDNA_VARIANT_UTS46 );
```

Comment:
Hmm, looking into it, I think we'd be better off calling `idn_to_utf8()` without specifying the `$flags` and `$variant` args, and instead letting it use the default values.

`$flags` defaults to `IDNA_DEFAULT` anyway, so this is really about the `$variant` argument.

As per the [docs](https://www.php.net/manual/en/function.idn-to-utf8.php), `INTL_IDNA_VARIANT_UTS46` is only available as of ICU 4.6.

However, the `intl` extension, which will be used if it exists, only [depends](https://www.php.net/manual/en/intl.requirements.php) on ICU 4.0+, until PHP 7.4 when it requires 50.1+. With Site Kit supporting PHP 5.6+, SK could well be installed on a system running an ICU version below 4.6.

Wanting to confirm the default `$variant` arg for older versions, I initially looked at the online [legacy docs](https://php-legacy-docs.zend.com/manual/php5/en/function.idn-to-utf8) for v5. However I found it a bit confusing because it refers to version 7 on there too. The [website itself](https://www.php.net/manual/php5.php) says the docs aren't perfect. So I dug through the PHP source, it looks like the default is [`INTL_IDNA_VARIANT_2003` in <=7.3](https://github.com/php/php-src/blob/php-7.3.33/ext/intl/idn/idn.c#L273-L277), and [`INTL_IDNA_VARIANT_UTS46` in 7.4+](https://github.com/php/php-src/blob/php-7.4.0/ext/intl/idn/idn.c#L247-L251).

So, in short it looks like we should be safe to stick with the defaults!

```suggestion
					$host_decoded = idn_to_utf8( $hostname );
```